lib/msf/core/auxiliary/manageengine_xnode/config.rb

```diff
@@ -0,0 +1,58 @@
+# -*- coding: binary -*-
+
+module Msf::Auxiliary::ManageengineXnode::Config
+  # Reads the configuration file for the current ManageEngine Xnode module in order to obtain the data repositories (tables) and fields (columns) to dump.
+  #
+  # @param config_file [String] String containing the full path to the configuration file to read.
+  # @return [Hash, Integer] Hash containing the data repositories (tables) and their fields (columns) to dump if reading the config file succeeded, error code otherwise
+  def grab_config(config_file)
+    # get the specified data respositories (tables) and fields (columns) to dump from the config file
+    return 1 unless File.exists? config_file
```

Comment:
@gwillcox-r7 I was unfamiliar with using enums, but I have attempted an implementation of them that I do agree makes the module more readable:
I have added the following in `config.rb`
```
module Msf::Auxiliary::ManageEngineXnode::Config
  CONFIG_FILE_DOES_NOT_EXIST = 1
  CANNOT_READ_CONFIG_FILE = 2
  DATA_TO_DUMP_EMPTY = 3
  DATA_TO_DUMP_WRONG_FORMAT = 4
```
and also this function
```
  def config_status
    Msf::Auxiliary::ManageEngineXnode::Config
  end
```
in the modules, the old `case data_to_dump`  statement has now been replaced with this one:
```
  case data_to_dump
      when config_status::CONFIG_FILE_DOES_NOT_EXIST
        fail_with(Failure::BadConfig, "Unable to obtain the Xnode data repositories to target from #{config_file} because this file does not exist. Please correct your 'CONFIG_FILE' setting or set 'DUMP_ALL' to true.")
      when config_status::CANNOT_READ_CONFIG_FILE
        fail_with(Failure::BadConfig, "Unable to read #{config_file}. Check if your 'CONFIG_FILE' setting is correct and make sure the file is readable and properly formatted.")
      when config_status::DATA_TO_DUMP_EMPTY
        fail_with(Failure::BadConfig, "The #{config_file} does not seem to contain any data repositories and fields to dump. Please fix your configuration or set 'DUMP_ALL' to true.")
      when config_status::DATA_TO_DUMP_WRONG_FORMAT
        fail_with(Failure::BadConfig, "Unable to obtain the Xnode data repositories to target from #{config_file}. Check if your 'CONFIG_DIR' setting is correct or set 'DUMP_ALL' to true.")
      end
 ```
I know passing the entire config module via `config_status` is not ideal, but I wanted to prevent having to use the full module name in the case statement because of how long it is. if you know of a better way to achieve the same, please let me know :)

Reply:
Approach looks reasonable to me, thanks for implementing this :)